### PR TITLE
[GPU] Fixup gemm attr

### PR DESCRIPTION
**Pull request by kealan-barbieri**
# Description

- Separate handling for edge case dim counts with non-trivial grouped attr.
- Fix scale batch offset computation
- Additional test cases
- Fix type initialization for zp in cases with `swap_ab`

Fixes # [MFDNN-13688](https://jira.devtools.intel.com/browse/MFDNN-13688)

# Checklist

## General

- [x] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit?
- [x] Have you formatted the code using clang-format?

### Bug fixes

- [x] Have you included information on how to reproduce the issue (either in a github issue or in this PR)?
- [x] Have you added relevant regression tests?

### CODE REVIEW

**Comment by kealan-barbieri** [comment]
make test
set test_scope=NIGHTLY
disable test_device_cpu
enable test_device_gpu
disable benchdnn_all
enable benchdnn_matmul

**Review by Simonsays095 — APPROVED** [review]

**Comment by kealan-barbieri** [comment]
make test perf-gpu
set primitive=gpu:gemm

File: tests/benchdnn/inputs/matmul/harness_matmul_3d_bcast [hunk 1]
```diff
@@ -12,11 +12,13 @@
 2x32x32:1x32x8
 26x32x32:1x32x65
 
-# Nontrivial src dims
+# Nontrivial per_tensor scales
 --reset
---dt=s8:u4:f16 --bia-dt=f16 --skip-impl=ref --attr-scales=src0:per_tensor:f16:1x32,src0:per_ocic:f16:1x32
+--dt=s8:u4:f16 --bia-dt=f16 --skip-impl=ref --attr-scales=src0:per_tensor:f16:1x32+wei:per_tensor:f16:32x1,src0:per_ocic:f16:1x32+wei:per_ocic:f16:32x1
```
Comment (dzarukin) [inline tests/benchdnn/inputs/matmul/harness_matmul_3d_bcast:17]:
```suggestion
--dt=s8:u4:f16 --bia-dt=f16 --skip-impl=ref --attr-scales=src0:per_tensor:f16:1x32+wei:per_tensor:f16:32x1,src0:per_tensor:f16:1x32+wei:per_tensor:f16:32x1
```

**Review by dzarukin — APPROVED** [review]

File: tests/benchdnn/inputs/matmul/harness_matmul_3d_bcast (hunk 1 above)
Comment (kealan-barbieri) [reply tests/benchdnn/inputs/matmul/harness_matmul_3d_bcast:17]: I want to test both kinds here, to make sure theyre different correctly.
